jetty-core/jetty-session/src/main/java/org/eclipse/jetty/session/AbstractSessionCache.java

```diff
@@ -107,7 +108,7 @@ public abstract class AbstractSessionCache extends ContainerLifeCycle implements
      * @return a new Session object
      */
     @Override
-    public abstract Session newSession(SessionData data);
+    public abstract ManagedSession newSession(SessionData data);
```

Comment:
Why do these all need to use the `ManagedSession` class and not the `Session` interface?

Reply:
Applications use `Session`.
Implementations of `Session` use `SessionManager`s and `SessionHandler`s and `SessionCache`s and `SessionData` to implement a `ManagedSession`, thus they communicate within themselves with `ManagedSession`.

If you are just a handler/servlet that wants to use a session, you just see/use `Session` interface. But if you are implementing/extending our style of concrete session you see/use `ManagedSession`